File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/GetKeyMetadataTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.wallet.WalletValueNotFoundException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class GetKeyMetadataTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testGetKeyMetadataWorks() throws Exception {
		Signus.setKeyMetadata(wallet, VERKEY, METADATA).get();
		String receivedMetadata = Signus.getKeyMetadata(wallet, VERKEY).get();
		assertEquals(METADATA, receivedMetadata);
	}

	@Test
	public void testGetKeyMetadataWorksForEmptyString() throws Exception {
		Signus.setKeyMetadata(wallet, VERKEY, "").get();
		String receivedMetadata = Signus.getKeyMetadata(wallet, VERKEY).get();
		assertEquals("", receivedMetadata);
	}

	@Test
	public void testGetKeyMetadataWorksForNoMetadata() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletValueNotFoundException.class));

		Signus.getKeyMetadata(wallet, VERKEY).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/SignusTests/GetKeyMetadataTest.cs
using Hyperledger.Indy.SignusApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{
    [TestClass]
    public class GetKeyMetadataTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestGetKeyMetadataWorks()
        {
            await Signus.SetKeyMetadataAsync(wallet, VERKEY, METADATA);
            var receivedMetadata = await Signus.GetKeyMetadataAsync(wallet, VERKEY);
            Assert.AreEqual(METADATA, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetKeyMetadataWorksForEmptyString()
        {
            await Signus.SetKeyMetadataAsync(wallet, VERKEY, string.Empty);
            var receivedMetadata = await Signus.GetKeyMetadataAsync(wallet, VERKEY);
            Assert.AreEqual(string.Empty, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetKeyMetadataWorksForNoMetadata()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletValueNotFoundException>(() =>
               Signus.GetKeyMetadataAsync(wallet, VERKEY)
           );
        }
    }
}
